src/transforms/api/_transform.py

```diff
@@ -226,6 +280,95 @@ def filesystem(self):
         )
 
 
+class LightweightTransformInput(TransformInput):
+    """LightweightTransformInput class, passed when using @lightweight decorator."""
+
+    def __init__(self, input_arg: Input):
+        super().__init__(input_arg)
+        self.branch = input_arg.branch
+        self.rid = input_arg.get_dataset_identity()["dataset_rid"]
+        self._path = input_arg.get_local_path_to_dataset()
+        del self.path
+
+    @property
+    def _parquet_files(self) -> List[Path]:
+        return list(Path(self._path).glob("**/*.parquet"))
+
+    @property
+    def _csv_files(self) -> List[Path]:
+        return list(Path(self._path).glob("**/*.csv"))
+
+    def dataframe(self):
+        raise NotImplementedError(
+            "Lightweight transforms do not support the dataframe method."
+        )
+
+    def path(self) -> str:
+        """Download the dataset's underlying files and return a Path to them."""
+        return self._path
+
+    def pandas(self) -> "pandas.DataFrame":
+        """A Pandas DataFrame containing the full view of the dataset."""
+        import pandas as pd
+
+        return (
+            pd.concat(map(pd.read_parquet, self._parquet_files), ignore_index=True)
+            if self._parquet_files
```

Comment:
Use walrus operator, so it searches for parquet files only once.

EDIT: I see that both methods get used multiple times, so switching _csv_files/_parquet_files to `@cached_property` could improve performance with big datasets.

```suggestion
            pd.concat(map(pd.read_parquet, parquet_files), ignore_index=True)
            if parquet_files := self._parquet_files
```

Reply:
Great idea!